tests/unittests/onnxImporterTest.cpp

```diff
@@ -132,3 +132,50 @@ TEST(onnx, importClip) {
     EXPECT_FLOAT_EQ(result.raw(i), expectedValues[i]);
   }
 }
+
+/// Test loading ReplaceNaN op from an ONNX model.
+/// Test with arg value = 1.0.
+TEST(onnx, importReplaceNaN) {
+  ExecutionEngine EE{BackendKind::Interpreter};
+  auto &mod = EE.getModule();
+  Function *F = mod.createFunction("main");
+
+  std::string netFilename("tests/models/onnxModels/replaceNaN.onnxtxt");
+
+  Context ctx;
+  Placeholder *output;
+  Tensor x(ElemKind::FloatTy, {3, 3});
+  auto XH = x.getHandle();
+
+  // Fill input by alternating between NAN and random values.
+  XH.randomize(-3.0, 3.0, mod.getPRNG());
+  for (size_t i = 0; i < XH.size(); i += 2) {
+    XH.raw(i) = NAN;
+  }
+
+  {
+    ONNXModelLoader onnxLD(netFilename, {"x"}, {&x.getType()}, *F);
+    output = onnxLD.getSingleOutput();
+    ctx.allocate(mod.getPlaceholders());
+
+    updateInputPlaceholdersByName(ctx, &mod, {"x"}, {&x});
+  }
+
+  auto *res = ctx.get(output);
+  EE.compile(CompilationMode::Infer, F, ctx);
+  EE.run();
+
+  auto result = res->getHandle();
+
+  // Output shape and input shape should match.
+  EXPECT_TRUE(result.dims().vec() == x.dims().vec());
+
+  // All NaNs in the input should be replaced with 1.0f and all other values
+  // should match.
+  for (size_t i = 0; i < result.size(); i++) {
+    auto resultValue = result.raw(i);
+    auto inputValue = XH.raw(i);
+
+    EXPECT_FLOAT_EQ(resultValue, std::isnan(inputValue) ? 1.0f : inputValue);
```

Comment:
very interesting. Can you `std::cout` resultValue and `std::isnan(inputValue) ? 1.0f : inputValue` before expect eq.

Reply:
Tried this locally and didn't find anything out of the ordinary:

```
...
Result value: 1
Expected value: 1
Result value: -2.18714
Expected value: -2.18714
Result value: 1
Expected value: 1
Result value: 2.01005
Expected value: 2.01005
Result value: 1
Expected value: 1
Result value: 2.81321
Expected value: 2.81321
Result value: 1
Expected value: 1
Result value: -1.6738
Expected value: -1.6738
Result value: 1
Expected value: 1
...
```

Reply:
Can you do the same on CI, so we could look at the output.